From 5a23fb9e583171d786a529a4eb9bd26de5b32167 Mon Sep 17 00:00:00 2001
From: ywan171 <yi.a.wang@intel.com>
Date: Thu, 16 May 2013 11:01:03 +0800
Subject: [PDK MR2 AOSP]:fix progress bar of the rtsp streaming from vlc server is not correct on MR2

BZ: 107186

The response of "PLAY" request from vlc rtsp  server does
not comply with the protocol. For a Non-live streaming, the form of
"range" property is "Range: npt=0-\r\n". In the method parsePlayResponse()
of MyHandler, the property of "Range" in this form could not be parsed
successfully, and the stream would be recognized as a live stream.
while in MR2,since it can get duration info from the SDP,the stream
would be not recognized as a live stream.The unconformity will lead
the wrong time stamp calculation.

Change-Id: Idc1fad17911ea1aa9c02f4b14e97661d63b7ce51
Signed-off-by: ywan171 <yi.a.wang@intel.com>
---
 media/libstagefright/rtsp/MyHandler.h |   12 +++++++-----
 1 files changed, 7 insertions(+), 5 deletions(-)

diff --git a/media/libstagefright/rtsp/MyHandler.h b/media/libstagefright/rtsp/MyHandler.h
index 8dd6176..8e3b6a3 100644
--- a/media/libstagefright/rtsp/MyHandler.h
+++ b/media/libstagefright/rtsp/MyHandler.h
@@ -1407,12 +1407,14 @@ struct MyHandler : public AHandler {
         AString val;
         CHECK(GetAttribute(range.c_str(), "npt", &val));
 
-        float npt1, npt2;
+        float npt1 = 0;
+        float npt2;
         if (!ASessionDescription::parseNTPRange(val.c_str(), &npt1, &npt2)) {
-            // This is a live stream and therefore not seekable.
-
-            ALOGI("This is a live stream");
-            return;
+            // check whether is it a live stream, live streaming is not seekable
+            if (isLiveStream(mSessionDesc)) {
+                ALOGI("This is a live stream");
+                return;
+            }
         }
 
         i = response->mHeaders.indexOfKey("rtp-info");
-- 
1.7.4.1

